lib/iris/_concatenate.py

```diff
@@ -742,6 +742,17 @@ def register(self, cube, axis=None, error_on_mismatch=False,
             # Declare the nominated axis of concatenation.
             self._axis = candidate_axis
 
+        if match:
+            # If the protocube dimension order is constant (indicating it was
```

Comment:
That's what I thought too... This file is checked by the coding standards tests too I think.